streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsRebalanceListener.java

```diff
@@ -50,6 +50,8 @@ public void onPartitionsAssigned(final Collection<TopicPartition> partitions) {
             log.error("Received error code {} - shutdown", streamThread.getAssignmentErrorCode());
             streamThread.shutdown();
         } else {
+            taskManager.handleRebalanceComplete();
```

Comment:
The ordering within consumer coordinator is the following:

1. assignor#onAssignment
2. the assignment gets updated internally
3. listener#onPartitionsAssigned

So we have to call `consumer.pause(all)` it inside 3) since otherwise the assignment() is still stale.

Reply:
oh boy...